bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/api/EtherRainCommunication.java

```diff
@@ -0,0 +1,243 @@
+                        break;
+                    case "ri":
+                        response.setLastActiveValue(Integer.parseInt(status));
+                        break;
+                    case "rn":
+                        response.setRainSensor(Integer.parseInt(status) == 1);
+                        break;
+                    default:
+                        logger.debug("Unknown response: {}", command);
+                }
+            }
+        }
+
+        return response;
+    }
+
+    public boolean commandIrrigate(int delay, int zone1, int zone2, int zone3, int zone4, int zone5, int zone6,
+            int zone7, int zone8) {
+        try {
+            sendGet("result.cgi?xi=" + delay + ":" + zone1 + ":" + zone2 + ":" + zone3 + ":" + zone4 + ":" + zone5 + ":"
+                    + zone6 + ":" + zone7 + ":" + zone8);
+        } catch (IOException e) {
+            logger.warn("Could not send irrigate command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public boolean commandClear() {
+        try {
+            sendGet("/result.cgi?xr");
+        } catch (IOException e) {
+            logger.warn("Could not send clear command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public boolean commandLogin() throws EtherRainException {
+        try {
+            sendGet("/ergetcfg.cgi?lu=" + ETHERRAIN_USERNAME + "&lp=" + password);
+        } catch (IOException e) {
+            logger.warn("Could not send clear command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public boolean commandLogout() {
+        try {
+            sendGet("/ergetcfg.cgi?m=o");
+        } catch (IOException e) {
+            logger.warn("Could not send logout command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    private List<String> sendGet(String command) throws IOException {
+        String url = "http://" + address + ":" + port + "/" + command;
+
+        ContentResponse response;
+
+        try {
+            response = httpClient.newRequest(url).timeout(HTTP_TIMEOUT, TimeUnit.SECONDS).send();
+            if (response.getStatus() != HttpURLConnection.HTTP_OK) {
+                logger.warn("Etherrain return status other than HTTP_OK : {}", response.getStatus());
+                return new LinkedList<String>();
+            }
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            logger.warn("Could not connect to Etherrain with exception: {}", e.getMessage());
+            return new LinkedList<String>();
+        }
+
+        return new BufferedReader(new StringReader(response.getContentAsString())).lines().collect(Collectors.toList());
+    }
+
+    private static EtherRainUdpResponse updBroadcast() {
```

Comment:
What if someone has more than one Etherrain device on their network? Can this handle it?

Reply:
Okay, this is a good find. I have added logic to fix this, but I can only test the case with one and zero etherrain's on the network.......I only have one. :)